include/sea_dsa/ShadowMem.hh

```diff
@@ -30,7 +49,24 @@ public:
 	    llvm::Pass &pass /* for dominatorTree and assumptionCache*/,
 	    bool splitDsaNodes = false, bool computeReadMod = false,
 	    bool memOptimizer = true, bool useTBAA = true);
+  
   bool runOnModule(llvm::Module &M);
+
+  // Return the offset associated to cell c. The offset can be
+  // adjusted to zero if horn-sea-dsa-split=false.
+  unsigned getOffset(const Cell &c) const;
+
+  ShadowMemInstOp getShadowMemOp(const llvm::CallInst &ci) const; 
+  
+  // Return cell associated to the shadow mem call instruction.  It
+  // can be null if ci is not a shadow mem call instruction.
+  const Cell* getShadowMemCell(const llvm::CallInst &ci) const;
```

Comment:
I don't remember if we pass `Cell` by value. If we do, this should be `llvm::Optional<Cell>`